### PR TITLE
Fix state not defined issue in #173

### PR DESCRIPTION
**Pull request by michaelize**
This PR fixes https://github.com/nuxt-community/nuxt-i18n/pull/173

The problem occurs when deploying with SSR to a static host, such as AWS S3, then access `https://example.com/en`. Since there is no matching file,  Nuxt loads the fallback file `https://example.com/200.html`.

Now due to there is no server rendered state value defined(https://nuxtjs.org/guide/vuex-store#modules-mode), client preserves the state without the i18n module state, which raised an exception later when trying to set the locale.

The fix is to set `{ preserveState: false }` by default. If need to pass value from the server, then set`preserveState` to true in the options, and create a module vuex file as described here: https://nuxtjs.org/guide/vuex-store#modules-mode

### CODE REVIEW

**Comment by priezz** [comment]
I would like to note, that #173 occures even when the SSR is not meant at all, i.e. in SPA nuxt mode.

**Comment by michaelize** [comment]
@priezz You are correct. This is not unique to SSR. It's how Vuex implements preserveState option. When preserveState is set to true, it skips adding state for the module since it assumes the state value already exists, as it can be passed from SSR.

Here's the explanation about Vuex preserveState; https://github.com/vuejs/vuex/issues/1130#issuecomment-356612119

**Comment by paulgv** [comment]
Hi and thanks for the PR!
Should I assume that merging this would fix the issue for both SSR and SPA modes?
Also, wouldn't it make more sense to set `preserveState` default value to `true` instead of `false`? Let me know.

Sorry for the delay in checking the PR.

**Comment by michaelize** [comment]
@paulgv Sorry for the late response.

TLDR: it works for both cases.

The plugin became broken due to a Nuxt change mentioned in one of the issues you posted here: https://github.com/nuxt/nuxt.js/issues/3757#issuecomment-422437601. Nuxt now overrides the store.registerModule method for universal apps after this PR https://github.com/nuxt/nuxt.js/pull/3909/files#diff-c1b82c7b8931d52d83e399952282cbc2R52

When preserveState is set to false, it simply initiates with local state. Since this module doesn't have any default server-rendered state, it should work fine in either case.

The reason @priezz is having this issue even when SSR is not being used is very likely due to the mode is set to universal. Pls let me know if this is not correct.

This problem was also identified on Nuxt repo: https://github.com/nuxt/nuxt.js/issues/4370

**Comment by priezz** [comment]
@michaelize Mode is set to 'spa' in my case.

**Comment by michaelize** [comment]
@priezz I couldn't reproduce this issue under spa mode. Can you create a sample repo?
